src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -23,7 +23,8 @@ public static partial class PlatformDetection
         // do it in a way that failures don't cascade.
         //
 
-        public static readonly bool IsInHelix = Environment.GetEnvironmentVariables().Keys.Cast<string>().Where(key => key.StartsWith("HELIX")).Any();
+        private static Lazy<bool> s_IsInHelix => new Lazy<bool>(() => Environment.GetEnvironmentVariables().Keys.Cast<string>().Any(key => key.StartsWith("HELIX")));
```

Comment:
Think you want a field here, i.e. `private static Lazy<bool> s_IsInHelix = ...`

s_IsInHelix is creating a new Lazy for each invocation of the getter, so there is no caching happening here.

Reply:
Good catch. We have the same problem in number of places actually.

https://github.com/dotnet/runtime/pull/68473 has the fix